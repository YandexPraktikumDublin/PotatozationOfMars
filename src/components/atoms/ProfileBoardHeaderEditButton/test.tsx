import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { ProfileBoardHeaderEditButton } from '.'

describe('<ProfileBoardHeaderEditButton />', () => {
  it('should renders correct <ProfileBoardHeaderEditButton />', () => {
    const onClick = jest.fn()

    const wrapper = shallow(<ProfileBoardHeaderEditButton onClick={onClick} />)

    wrapper.simulate('click')

    expect(onClick).toHaveBeenCalledTimes(1)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
