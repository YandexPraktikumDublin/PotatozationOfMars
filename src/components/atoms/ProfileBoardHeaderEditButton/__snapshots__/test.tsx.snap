// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileBoardHeaderEditButton /> should renders correct <ProfileBoardHeaderEditButton /> 1`] = `
<button
  className="absolute top-0 right-0"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <svg
    height="20"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      className="fill-primary dark:fill-white"
    >
      <path
        d="M12.652 3.072L15.724 0 20 4.275l-3.072 3.073-4.276-4.276zM0 20l5.264-1.514-3.75-3.75L0 20zM6.477 17.799L2.2 13.523l9.501-9.5 4.276 4.275-9.501 9.5z"
      />
    </g>
  </svg>
</button>
`;
